docker_util.py

```diff
@@ -0,0 +1,84 @@
+import json
+import subprocess
+
+from verbose_print import print_v
+
+
+class ImageNotFoundError(Exception):
+    def __init__(self, image):
+        self.image = image
+        self.message = f'The image {image} was not found.'
+
+    def __str__(self):
+        return self.message
+
+
+def tag_image(source, target, is_dry_run):
+    print(f'Tagging image {source} as {target}')
+    tag_command = ('docker', 'tag', source, target)
+    if is_dry_run:
+        print_v(' '.join(tag_command))
+        print('Image was not actually tagged since this is a dry run')
+    else:
+        _check_output_and_print(tag_command)
+
+
+def pull_image(image):
+    pull_command = ('docker', 'pull', image)
+    try:
+        _check_output_and_print(pull_command)
+    except subprocess.CalledProcessError as e:
+        stderr = e.stderr.decode()
+        if image in stderr and 'not found' in stderr:
```

Comment:
this might be fragile, what if the output of the `docker` command changes -- it's probably safe to just return the status code here and if it's nonzero assume that the pull failed and proceed as normal.

Reply:
True expecting some text to appear is very fragile. I mostly wanted to surface errors that weren't related to an image actually not being found (i. e. docker not being installed or the command/image name being malformed).

But I guess it's fine, stderr will still show up to show what's wrong and things will just blow up elsewhere